python/cudf/cudf/tests/test_seriesmap.py

```diff
@@ -73,3 +73,12 @@ def test_series_map_callable_numeric_random_dtype_change():
     expect = np.array(data, dtype=float)
     got = out.to_array()
     np.testing.assert_array_equal(expect, got)
+
+
+def test_series_map_non_unique_index():
+    # Test for changing the out_dtype using applymap
+
+    gd2 = cudf.Series([1, 2, 3, 4, np.nan])
+
+    with pytest.raises(ValueError):
+        assert gd2.map(cudf.Series(["a", "b", "c"], index=[1, 1, 2]))
```

Comment:
Instead of hardcoding the exception here, can you use [`assert_exceptions_equal`](https://github.com/rapidsai/cudf/blob/8aef96602478d733969011fecc73967ce5f95f97/python/cudf/cudf/tests/utils.py#L121) to ensure we're raising the same exception as Pandas?

Reply:
Sounds good!

Reply:
It looks like pandas raises a custom `InvalidIndexError`. Should I add this error for us? It doesn't look like we have it or use it anywhere else at the moment.

Reply:
@galipremsagar @shwina what have we been doing thus far in these situations?

Reply:
We allow reindexing with duplicate index just like pandas does. But error on non-unique index at specific places:

https://github.com/rapidsai/cudf/blob/094c32547bf384206d2a2f63858f07e6b30e61f0/python/cudf/cudf/core/series.py#L4189-L4193

Reply:
In my code, I had raised a value error similar to both of those cases. It just doesn't pass the `assert_exceptions_equal` test because the base errors are different. Is there a way to get around this?

Reply:
you can pass `check_exception_type=False` and send the desired error message via `expected_error_message`

Reply:
Great! Thanks, I'll do that :D